### ISSUE TITLE
Bug - Can not add existing works to existing collections.

### ISSUE THREAD

**Issue by crisr15** [scientist-softserv/palni-palci#430]

**Comment by bkiahstroud** [comment]
Currently running this script on production. It should resolve the issue across all tenants: 

```ruby
Account.find_each do |a|
  AccountElevator.switch!(a.cname)

  Collection.find_each do |c|
    c.reindex_extent = Hyrax::Adapters::NestingIndexAdapter::LIMITED_REINDEX
    pt = Hyrax::PermissionTemplate.find_or_create_by!(source_id: c.id)

    pt.access_grants.find_or_create_by!(
      access: Hyrax::PermissionTemplateAccess::MANAGE,
      agent_type: Hyrax::PermissionTemplateAccess::GROUP,
      agent_id: Ability.admin_group_name
    )

    pt.reset_access_controls_for(collection: c, interpret_visibility: true)
  end;nil
end;nil
```

**Comment by bkiahstroud** [comment]
[This PR](https://github.com/samvera/hyku/pull/1927) should prevent this from happening in upstream Hyku. We should not need to worry about backporting this to pals since it shouldn't happen again now that we've run [this script](https://github.com/scientist-softserv/palni-palci/issues/430#issuecomment-1459105812) on the prod server

**Comment by bkiahstroud** [comment]
"Good" collection (the admin group has manage access to the collection) 

<img width="1102" alt="Screenshot 2023-03-07 at 3 22 12 PM" src="https://user-images.githubusercontent.com/32469930/223600528-12a76209-e223-4b27-8972-eb1480ad730a.png">

"Bad" collection (the admin group does not have manage access to the collection) 

<img width="1102" alt="Screenshot 2023-03-07 at 3 19 57 PM" src="https://user-images.githubusercontent.com/32469930/223600549-1ef9725e-57b7-4129-94da-0b11500a08a8.png">

**Comment by ndroark** [comment]
Is this effective in production already or does it need testing in staging first?

**Comment by bkiahstroud** [comment]
> Is this effective in production already or does it need testing in staging first?

@ndroark this is effective in production now (it was an issue with the data)

**Comment by crisr15** [comment]
Per Kiah's note above moving to Client Verification.

**Comment by bkiahstroud** [comment]
> it may actually be related to the `page` param. If you navigate to any page of results after the first, the "you do not have access" message displays consistently 

I looked at this with @laritakr. We determined this is almost certainly a bug in Hyrax. To corroborate this, I was able to replicate the [same issue on UTK staging](https://dc.utk-hyku-staging.notch8.cloud/dashboard/my/works/page/3?add_works_to_collection=d354d217-83c9-4194-a765-2bba18a57393&add_works_to_collection_label%5B%5D=Ed+Gamble+Cartoon+Collection&locale=en), which is running Hyrax `v3.5.0`. 

> when attempting to add them to a collection... The page just spins for 30-45 seconds while submitting the form, ending in a 504 error 

TL;DR - I think this may also be a Hyrax bug. It seems like the collection and all its children are being reindexed, which takes a long time. 

```ruby
resource.member_of_collection_ids
=> []
Hyrax.persister.resource_factory.from_resource(resource: resource)
=> # takes milliseconds

resource.member_of_collection_ids
=> ["3efc445a-42f3-442b-9581-ae1286f093f6"]
Hyrax.persister.resource_factory.from_resource(resource: resource)
=> # take several seconds to several minutes (reindexes collection and all members) 
```

**Comment by DiemBTran** [comment]
**This passes SoftServ QA:** See attached screenshots

tested on:
- sample CSV to make 80 simple works: 
[many_simple.csv](https://github.com/scientist-softserv/palni-palci/files/11143299/many_simple.csv)
- step 2's collection: https://qa-13.commons-archive.org/collections/f2fa91d2-2abd-4abd-98d2-b1ded8ff01fd
- step 3's existing work: https://qa-13.commons-archive.org/concern/generic_works/e44ad6bd-f9cd-4e86-a73e-68badc04e87b?locale=en

<details><summary>I can add existing works to existing collections without error</summary>

![work in non-mediated deposit ID e44ad6bd-f9cd-4e86-a73e-68badc04e87b Hyku Commons 2023-04-03 at 5 11 02 PM](https://user-images.githubusercontent.com/29311858/229653795-6823b562-1c11-4821-8311-1ec136ebf80f.jpg)
![Can not add existing works to existing collections #430 Hyku Commons 2023-04-03 at 5 10 53 PM](https://user-images.githubusercontent.com/29311858/229653801-5d45874d-b373-4891-bf76-99e9d2c7cc64.jpg)
![Can not add existing works to existing collections #430 Hyku Commons 2023-04-03 at 4 32 36 PM](https://user-images.githubusercontent.com/29311858/229653804-42e2ff8c-79ad-4bdb-beaa-199eb91109a5.jpg)
</details>

**Comment by bkiahstroud** [comment]
Sister Hyrax ticket: https://github.com/samvera/hyrax/issues/6006
